### PR TITLE
fix #241 call initChart when config.options are registered so that they are reflected in directives

### LINKED ISSUE #241: options are not reflected in directives
options object is watched for change.
however, highcharts-ng does not reflects the values when it is used in directives.

In the [example](http://jsfiddle.net/yujiosaka/bqvpe2mj/2/), line has dash style as specified in options,
but it is not reflected when it's drawn.

### PR DESCRIPTION
**Pull request by yujiosaka**
fix #241

### CODE REVIEW

**Comment by yujiosaka** [comment]
"config.options" is watched for change.
however, it does not call `initChart()` in the initial change when it's used in directives.
therefore, high charts-ng does not reflect the initial value.

this is illustrated in the [example](http://jsfiddle.net/yujiosaka/bqvpe2mj/2/)

please read #241 as well.

File: src/highcharts-ng.js [hunk 1]
```diff
@@ -324,11 +324,12 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
           }, true);
         });
         scope.$watch('config.options', function (newOptions, oldOptions, scope) {
-          //do nothing when called on registration
-          if (newOptions === oldOptions) return;
           initChart();
-          processSeries(scope.config.series);
-          chart.redraw();
+          //do not redraw when called on registration
+          if (newOptions === oldOptions) {
```
Comment (pablojim) [inline src/highcharts-ng.js]: Surely this WILL redraw when called on registration

File: src/highcharts-ng.js [hunk 2]
```diff
@@ -324,11 +324,12 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
           }, true);
         });
         scope.$watch('config.options', function (newOptions, oldOptions, scope) {
-          //do nothing when called on registration
-          if (newOptions === oldOptions) return;
           initChart();
```
Comment (pablojim) [inline src/highcharts-ng.js]: initChart will be then called twice for a new chart

**Comment by yujiosaka** [comment]
@pablojim 

thanks for your comments. would you check and confirm that `initChart` is not called on registration *when it's used in directives*, as illustrated in this [example](http://jsfiddle.net/yujiosaka/bqvpe2mj/2/)?

i also created another [example](http://jsfiddle.net/n4ttnwhb/1/) of how i circumvent this problem by adding the following line:

```
scope.chartConfig.options.rand = Math.random();
```

In this example, the dash style of plotOptions is working.
I would like to see highcharts-ng working on directives without this ugly workaround.

**Comment by pablojim** [comment]
By using a controller function this seems to work: http://jsfiddle.net/Lnd8skdp/

Does this help?

**Comment by ccemeraldeyes** [comment]
+1 this is a HUGE issue for me as well.

**Comment by pablojim** [comment]
Closing post 1.0.0 release

**Pull request closed without merging**